File: Microservices/IoT.House.Automation.Microservices.Arduino/IoT.House.Automation.Microservices.Arduino.Domain/Models/Arduino.cs
using System;
using System.Collections.Generic;
using System.Net;
using System.Text;

namespace IoT.House.Automation.Microservices.Arduino.Domain.Models
{
    public class Arduino
    {
        public Guid UniqueIdentifier { get; set; }
        public string Name { get; set; }
        public IPAddress IP { get; set; }
        public int Port { get; set; }
        public IEnumerable<Event> Events { get; set; }
    }
}

File: Microservices/IoT.House.Automation.Microservices.Mobile/domain/src/main/java/iot/house/automation/microservices/mobile/domain/models/arduino/Arduino.java
package iot.house.automation.microservices.mobile.domain.models.arduino;

import iot.house.automation.microservices.mobile.domain.models.arduino.events.Event;
import lombok.Getter;

import java.util.List;
import java.util.UUID;

@Getter
public class Arduino {
    private final UUID uniqueIdentifier;
    private final String name;
    private final List<Event> events;

    public Arduino(UUID uniqueIdentifier, String name, List<Event> events) {
        this.uniqueIdentifier = uniqueIdentifier;
        this.name = name;
        this.events = events;
    }

    public boolean isValid(){
        return this.uniqueIdentifier != null
                && this.name != null
                && this.events != null
                && this.events.size() > 0;
    }
}